arch/risc-v/src/common/riscv_pmp.c

```diff
@@ -115,42 +108,54 @@ typedef struct pmp_entry_s pmp_entry_t;
 static bool pmp_check_region_attrs(uintptr_t base, uintptr_t size,
                                    uintptr_t type)
 {
-  /* Check that the size is not too small */
+  switch (type)
+  {
+    case PMPCFG_A_TOR:
 
-  if ((type != PMPCFG_A_TOR) && (size < MIN_BLOCK_SIZE))
-    {
-      return false;
-    }
+      /* For TOR any size is good, but alignment requirement stands */
 
-  /* Check that the base address is aligned properly */
+      if ((base & 0x03) != 0)
+        {
+          return false;
+        }
 
-  if ((base & BLOCK_ALIGN_MASK) != 0)
-    {
-      return false;
-    }
+      break;
 
-  /* Check that the size is aligned properly */
+    case PMPCFG_A_NA4:
 
-  if ((size & BLOCK_ALIGN_MASK) != 0)
-    {
-      return false;
-    }
+      /* For NA4 only size 4 is good, and base must be aligned */
+
+      if ((base & 0x03) != 0 || size != 4)
+        {
+          return false;
+        }
 
-  /* Perform additional checks on base and size for NAPOT area */
+      break;
 
-  if (type == PMPCFG_A_NAPOT)
-    {
-      /* Get the power-of-two for size, rounded up */
+    case PMPCFG_A_NAPOT:
+      {
+        /* For NAPOT, both base and size must be properly aligned */
 
-      uintptr_t pot = LOG2_CEIL(size);
+        if ((base & 0x07) != 0 || size < 8)
+          {
+            return false;
+          }
 
-      if ((base & ((UINT64_C(1) << pot) - 1)) != 0)
-        {
-          /* The start address is not properly aligned with size */
+        /* Get the power-of-two for size, rounded up */
```

Comment:
```suggestion
          /* Get the power-of-two for size, rounded up */

Reply:
The indent is correct.

I think the unified diff shows the change in a pretty confusing way, I suggest you change to split view which shows the change correctly!